com.playeveryware.eos/Runtime/Core/Config/ProductConfig.cs

```diff
@@ -114,6 +141,86 @@ protected override bool NeedsMigration()
         
         protected ProductConfig() : base("eos_product_config.json") { }
 
+        protected override void OnReadCompleted()
+        {
+            // This tracks whether there is a single deployment defined.
+            _deploymentDefined = Environments.IsDeploymentDefined;
+        }
+
+        // This compile conditional is here because the OnWriteCompleted method
+        // is only defined in the editor - so it can only be overriden if in the
+        // editor.
+#if UNITY_EDITOR
+
+        protected override void BeforeWrite()
+        {
+            // If there is one deployment, and one sandbox, then make sure they
+            // are linked to each other if the sandbox is not empty.
+            // But only do this if they have been newly added
+            if (!_deploymentDefined && Environments.Deployments.Count ==1 && Environments.Sandboxes.Count == 1 && !Environments.Sandboxes[0].Value.IsEmpty)
+            {
+                Environments.Deployments[0].Value.SandboxId = Environments.Sandboxes[0].Value;
+            }
+        }
+
+        protected override void OnWriteCompleted()
+        {
+            // If when the config was last read there was a deployment defined,
+            // or there is not now one defined - then there is no need to try
+            // and set the deployment values for each platform config.
+            if (_deploymentDefined || !Environments.IsDeploymentDefined)
+            {
+                return;
+            }
+
+            // Select the first defined deployment as the deployment to set
+            // platform configs to use.
+            Named<Deployment> deploymentToSetPlatformsTo = Environments.Deployments[0];
+
+            List<PlatformManager.Platform> platformConfigsUpdated = new();
+
+            // For each platform for which configuration can be done
+            foreach (var platform in PlatformManager.ConfigurablePlatforms)
+            {
+                // If the PlatformConfig could not be retrieved, continue to the
+                // next.
+                if (!PlatformManager.TryGetConfig(platform, out PlatformConfig config))
+                {
+                    continue;
+                }
+
+                // If the config already has a completely defined deployment,
+                // then do not override, and move to the next platform config
+                if (config.deployment.IsComplete)
+                {
+                    continue;
+                }
+
+                // Add to the list of platform configs that have been updated
+                platformConfigsUpdated.Add(platform);
+
+                // Set the deployment.
+                config.deployment = deploymentToSetPlatformsTo.Value;
```

Comment:
This fails if `deploymentToSetPlatformsTo` is null.

Reply:
If `deploymentToSetPlaytformsTo` is not defined - then the function will not get to that point.

Reply:
`deploymentToSetPlatformsTo` was null due to a boolean logic error; that has been fixed in another discussion.